common/src/main/java/com/wynntils/screens/maps/SeaskipperMapScreen.java

```diff
@@ -0,0 +1,522 @@
+                        Component.literal("[>] ")
+                                .withStyle(ChatFormatting.LIGHT_PURPLE)
+                                .append(Component.translatable("screens.wynntils.seaskipperMapGui.buyBoat.name")),
+                        Component.translatable("screens.wynntils.seaskipperMapGui.buyBoat.description")
+                                .withStyle(ChatFormatting.GRAY))));
+    }
+
+    @Override
+    public void doRender(PoseStack poseStack, int mouseX, int mouseY, float partialTick) {
+        updateMapCenterIfDragging(mouseX, mouseY);
+
+        RenderSystem.setShaderColor(1f, 1f, 1f, 1f);
+
+        RenderSystem.enableDepthTest();
+
+        renderMap(
+                poseStack,
+                Managers.Feature.getFeatureInstance(GuildMapFeature.class)
+                        .renderUsingLinear
+                        .get());
+
+        RenderUtils.enableScissor(
+                (int) (renderX + renderedBorderXOffset), (int) (renderY + renderedBorderYOffset), (int) mapWidth, (int)
+                        mapHeight);
+
+        renderPois(poseStack, mouseX, mouseY);
+
+        renderCursor(
+                poseStack,
+                1.5f,
+                Managers.Feature.getFeatureInstance(GuildMapFeature.class)
+                        .pointerColor
+                        .get(),
+                Managers.Feature.getFeatureInstance(GuildMapFeature.class)
+                        .pointerType
+                        .get());
+
+        RenderUtils.disableScissor();
+
+        renderBackground(poseStack);
+
+        renderCoordinates(poseStack, mouseX, mouseY);
+
+        renderMapButtons(poseStack, mouseX, mouseY, partialTick);
+
+        renderHoveredSeaskipperDestination(poseStack);
+    }
+
+    private void renderPois(PoseStack poseStack, int mouseX, int mouseY) {
+        Vec3 playerLocation = McUtils.player().position();
+
+        for (Poi poi : seaskipperPois) {
+            SeaskipperPoi seaskipperPoi = (SeaskipperPoi) poi;
+
+            if (seaskipperPoi.isPlayerInside(playerLocation)) {
+                currentPoi = poi;
+                break;
+            }
+        }
+
+        if (renderAllDestinations) {
+            renderPois(
+                    seaskipperPois,
+                    poseStack,
+                    BoundingBox.centered(mapCenterX, mapCenterZ, width / currentZoom, height / currentZoom),
+                    1,
+                    mouseX,
+                    mouseY);
+        } else {
+            availableDestinations = new ArrayList<>();
+
+            for (SeaskipperDestinationItem destinationItem : destinations.keySet()) {
+                String destination = destinationItem.getDestination();
+
+                for (Poi poi : seaskipperPois) {
+                    if (poi.getName().equals(destination)) {
+                        availableDestinations.add(poi);
+                        break;
+                    }
+                }
```

Comment:
Seems a good candidate to using streams to make this a bit shorter/readable

Reply:
First time using streams, so might not be perfect but I have given it an attempt that does work, let me know if it can be improved further